pipeline/src/constructors/remake_latent_model.jl

```diff
@@ -3,91 +3,100 @@ Constructs and returns a latent model based on the provided `inference_config` a
+For Direct Infections process `log(I_t)` in a single time step a fluctuation of 0.025 is not unexpected.
+
+The autocorrelation is expected to be 0.1.
+"""
+function _make_target_std_and_autocorr(igp; stationary::Bool)
+    if igp == Renewal
+        return stationary ? (0.75, 0.1) : (0.025, 0.1)
+    elseif igp == ExpGrowthRate
+        return stationary ? (0.1, 0.1) : (0.005, 0.1)
+    elseif igp == DirectInfections
+        return stationary ? (1.0, 0.5) : (0.025, 0.1)
+    end
+end
+
+"""
+Constructs and returns a latent model based on an approximation to the specified target standard deviation and autocorrelation.
+
+NB: The stationary variance of an AR(1) process is given by `σ² = σ²_ε / (1 - ρ²)` where `σ²_ε` is the variance of the noise and `ρ` is the autocorrelation.
+The approximation here are based on `E[1/(1 - ρ²)`] ≈ 1 / (1 - E[ρ²])` which only holds for fairly tight distributions of `ρ`.
+However, for priors this should get the expected order of magnitude.
+
+# Models
+- `"diff_ar"`: Constructs a `DiffLatentModel` with an autoregressive (AR) process.
+- `"ar"`: Constructs an autoregressive (AR) process.
+- `"rw"`: Constructs a random walk (RW) process.
+
+"""
+function _implement_latent_process(
+        target_std, target_autocorr, latent_model_name, pipeline; beta_eff_sample_size = 10)
+    prior_dict = make_model_priors(pipeline)
+    damp_prior = Beta(target_autocorr * beta_eff_sample_size,
+        (1 - target_autocorr) * beta_eff_sample_size)
+    corr_corrected_noise_std = HalfNormal(target_std * sqrt(1 - target_autocorr^2))
+    noise_std = HalfNormal(target_std)
+    init_prior = prior_dict["transformed_process_init_prior"]
     if latent_model_name == "diff_ar"
-        if igp == Renewal
-            ar = AR(damp_priors = [prior_dict["damp_param_prior"]],
-                std_prior = HalfNormal(0.05 / tscale),
-                init_priors = [prior_dict["transformed_process_init_prior"]])
-            diff_ar = DiffLatentModel(;
-                model = ar, init_priors = [prior_dict["transformed_process_init_prior"]])
-            return diff_ar
-        elseif igp == ExpGrowthRate
-            ar = AR(damp_priors = [prior_dict["damp_param_prior"]],
-                std_prior = HalfNormal(0.005 / tscale),
-                init_priors = [prior_dict["transformed_process_init_prior"]])
-            diff_ar = DiffLatentModel(;
-                model = ar, init_priors = [prior_dict["transformed_process_init_prior"]])
-            return diff_ar
-        elseif igp == DirectInfections
-            ar = AR(damp_priors = [Beta(9, 1)],
-                std_prior = HalfNormal(0.05 / tscale),
-                init_priors = [prior_dict["transformed_process_init_prior"]])
-            diff_ar = DiffLatentModel(;
-                model = ar, init_priors = [prior_dict["transformed_process_init_prior"]])
-            return diff_ar
-        end
+        _ar = AR(damp_priors = [damp_prior],
```

Comment:
this is a dupe of the ar specification below

Reply:
I was avoiding construction if latent process was "rw"

Reply:
But probably not worth it

Reply:
Not relevant with latest change